### PR TITLE
MC-18515: Update commitment API docs

### PR DESCRIPTION
**Pull request by joelplourde4**
#### Changes made
- Add the Update commitment endpoint to the API docs

#### Related PRs
https://github.com/cloudops/cloudmc-ui/pull/2750
https://github.com/cloudops/cloudmc-core/pull/2421

### CODE REVIEW

File: source/includes/administration/_commitments.md [hunk 1]
```diff
@@ -467,4 +467,71 @@ Attributes | &nbsp;
...
+```
+
+<!-------------------- UPDATE COMMITMENT -------------------->
+### Update commitment
+
+`PUT /commitments/:id`
+
+Update the attributes of a commitment for an organization
+
+A commitment that is applied can be updated. Meaning, that since an invoice hasn't been issued to the customer, all of the attributes of the commitment can be updated.
```
Comment (DMurray21) [inline source/includes/administration/_commitments.md]:
```suggestion
If an invoice hasn't been issued to the customer, all of the attributes of the commitment can be updated.
```

File: source/includes/administration/_commitments.md [hunk 2]
```diff
@@ -467,4 +467,71 @@ Attributes | &nbsp;
...
+
+A commitment that is pre-paid can be updated, **but** it will be terminated and a new commitment will be created with the updated attributes and will be effective at the start at the end of the current billing cycle.
+
+Updating a commitment always requires a reprocess from the start of the current billing cycle, expect if only the name has been updated.
```
Comment (DMurray21) [inline source/includes/administration/_commitments.md]: I don't think mentioning the reprocess in the API docs is worthwhile. It doesn't impact how the user interacts with the API.

File: source/includes/administration/_commitments.md [hunk 3]
```diff
@@ -467,4 +467,71 @@ Attributes | &nbsp;
...
+```
+
+Attributes | &nbsp;
+---- | ---
+`id`<br/>*string* | The identifier of the commitment.
+`name`<br/>*string* | The name of the commitment.
+`organization`<br/>*Organization* | The organization that is associated to the commitment.
+`reseller`<br/>*Organization* | The organization that owns the commitment.
+`committedProducts`<br/>*Array[CommittedProduct]* | The list of products, commitment levels and negotiated price associated to the commitment.
+`pricingMethod`<br/>*enum* | The pricing method of the commitment. Possible values are: `FIXED_PRICE` and `UTILITY_DISCOUNT`. Used to define how the effective price of the commitment is calculated.
```
Comment (DMurray21) [inline source/includes/administration/_commitments.md:532]: We'll need to add the tax code here as well once @lniemann-co merges his PR

**Review by DMurray21 — CHANGES_REQUESTED** [review]
Good stuff! Minor comments

File: source/includes/administration/_commitments.md (hunk 1 above)
Comment (joelplourde4) [reply source/includes/administration/_commitments.md]: Done 130920581335d8572ed8f1c5ff2ebb5dd6004ad0

File: source/includes/administration/_commitments.md (hunk 2 above)
Comment (joelplourde4) [reply source/includes/administration/_commitments.md]: Done 130920581335d8572ed8f1c5ff2ebb5dd6004ad0

File: source/includes/administration/_commitments.md (hunk 3 above)
Comment (joelplourde4) [reply source/includes/administration/_commitments.md:532]: Pre-emptively done 130920581335d8572ed8f1c5ff2ebb5dd6004ad0

**Review by DMurray21 — APPROVED** [review]

**Pull request merged**